modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java

```diff
@@ -942,7 +943,10 @@ private void exec() {
                         message = (BMessage) sf.refRegs[i];
                     }
                     context.setError(null);
-                    context.getBalCallback().done(message != null ? message.value() : null);
+                    // this should be fixed once JMS transactions are available as it requires callback
+                    if (((DefaultBalCallback) context.getBalCallback()).getParentCallback() != null) {
+                        context.getBalCallback().done(message != null ? message.value() : null);
```

Comment:
Talked to Hasitha and got to know that context.getBalCallback() could be null. We need to check that too.